wasip1/lookup_wasip1_getaddrinfo.go

```diff
@@ -0,0 +1,104 @@
+//go:build wasip1 && getaddrinfo
+
+package wasip1
+
+import (
+	"context"
+	"errors"
+	"net"
+	"os"
+	"strconv"
+)
+
+func dialResolverNotSupported(ctx context.Context, network, address string) (net.Conn, error) {
+	// The net.Resolver type makes a call to net.DialUDP to determine which
+	// resolved addresses are reachable, which does not go through its Dial
+	// hook. As a result, it is unusable on GOOS=wasip1 because it fails
+	// even when the Dial function is set because WASI preview 1 does not
+	// have a mechanism for opening UDP sockets.
+	//
+	// Instead of having (often indirect) use of the net.Resolver crash, we
+	// override the Dial function to error earlier in the resolver lifecycle
+	// with an error which is more explicit to the end user.
+	return nil, errors.New("net.Resolver not supported on GOOS=wasip1")
+}
+
+func init() {
+	net.DefaultResolver.Dial = dialResolverNotSupported
+}
```

Comment:
What do you think about removing this entirely: always have the net.Resolver configured, only use the build tag to control the implementation of `lookupAddr` used by this package?

Reply:
Good idea. See cc815d33ca6c1243b6bf9333c05de20a0731c6e4